tests/log.rs

```diff
@@ -0,0 +1,105 @@
+extern crate glib;
+
+use glib::*;
+
+use std::sync::{Arc, Mutex};
+
+#[derive(Default)]
+struct Counters {
+    criticals: usize,
+    warnings: usize,
+    messages: usize,
+    infos: usize,
+    debugs: usize,
+}
+
+fn assert_counts(
+    count: &Arc<Mutex<Counters>>,
+    criticals: usize,
+    warnings: usize,
+    messages: usize,
+    infos: usize,
+    debugs: usize,
+) {
+    let count = count.lock().expect("failed to lock 1");
+    assert_eq!(count.criticals, criticals);
+    assert_eq!(count.warnings, warnings);
+    assert_eq!(count.messages, messages);
+    assert_eq!(count.infos, infos);
+    assert_eq!(count.debugs, debugs);
+}
+
+#[test]
+fn check_log_handlers() {
+    //
+    // log_set_default_handler check part
```

Comment:
Make sure to set the fatal_mask here explicitly so that criticals/warnings are not fatal :) Otherwise the environment might contain `G_DEBUG=fatal_warnings` and things would simply abort on the first critical/warning.